transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -723,7 +747,17 @@ protected void doClose() throws Exception {
 
     @Override
     protected void doBeginRead() throws Exception {
-        // NOOP
+        // read from tempInboundMessages and fire channelRead.
+        if (!tempInboundMessages().isEmpty()) {
+            Object msg;
+            while ((msg = tempInboundMessages().poll()) != null) {
+                ChannelPipeline p = pipeline();
+                p.fireChannelRead(msg);
+            }
```

Comment:
please use the same style we use everywhere else in netty:

```
for (;;) {
    Object msg = tempInboundMessages().poll()
    if (msg == null) {
        break;
    }
    pipeline.fireChannelRead(msg);
}
....
```

Reply:
done.